src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -137,6 +137,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DocumentationFile />
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(PublishDocumentationFile)' == ''">
+    <PublishDocumentationFile Condition="'$(GenerateDocumentationFile)' == 'true'">true</PublishDocumentationFile>
```

Comment:
The situation currently is: if you haven an app and library as `ProjectReference`, the documentation file of the library is published with the app, but not the app's own documentation.
That's the main reason why i chose it to be on by default.